File: app/api/schemas/telemetry.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class TelemetryParameterInput(BaseModel):
    name: str
    value: float


class TelemetryEventCreate(BaseModel):
    satellite_id: UUID
    timestamp: datetime
    parameters: list[TelemetryParameterInput]


class TelemetryEventBatchCreate(BaseModel):
    events: list[TelemetryEventCreate]


class TelemetryHistoryResponse(BaseModel):
    timestamp: datetime
    value: float

File: frontend/src/types/telemetry.ts
export interface TelemetryPoint {

  timestamp: string

  timestampMs: number

  value: number

}


export interface TelemetryParameter {

  name: string

  value: number

}


export interface TelemetryEvent {

  event_id: string

  satellite_id: string

  timestamp: string

  parameters: TelemetryParameter[]

}